packages/common-integration-tests/src/auth-client.ts

```diff
@@ -336,6 +351,207 @@ export function runAuthClientTests(
       expect(getResponse).toBeInstanceOf(CacheGet.Miss);
     });
   });
+
+  describe('Fine grained authorization scenarios', () => {
+    const FGA_CACHE_1 = 'fga-' + v4();
+    const FGA_CACHE_2 = 'fga-' + v4();
+    const FGA_TOPIC_1 = 'topic-1';
+    const FGA_TOPIC_2 = 'topic-2';
+    const FGA_CACHE_1_KEY = 'foo';
+    const FGA_CACHE_1_VALUE = 'FOO';
+    const FGA_CACHE_2_KEY = 'bar';
+    const FGA_CACHE_2_VALUE = 'BAR';
+
+    const trivialHandlers: SubscribeCallOptions = {
+      onError: () => {
+        return;
+      },
+      onItem: () => {
+        return;
+      },
+    };
+
+    // Setup 2 caches, and 2 topics on the first cache.
+    beforeAll(async () => {
+      expect(
+        await sessionTokenCacheClient.createCache(FGA_CACHE_1)
+      ).toBeInstanceOf(CreateCache.Success);
+      expect(
+        await sessionTokenCacheClient.createCache(FGA_CACHE_2)
+      ).toBeInstanceOf(CreateCache.Success);
+      expect(
+        await sessionTokenCacheClient.set(FGA_CACHE_1, 'foo', 'FOO', {ttl: 600})
+      ).toBeInstanceOf(CacheSet.Success);
+      expect(
+        await sessionTokenCacheClient.set(FGA_CACHE_2, 'bar', 'BAR', {ttl: 600})
+      ).toBeInstanceOf(CacheSet.Success);
+      expect(
+        await sessionTokenTopicClient.publish(
+          FGA_CACHE_1,
+          FGA_TOPIC_1,
+          'it is a nice day today!'
+        )
+      ).toBeInstanceOf(TopicPublish.Success);
+      expect(
+        await sessionTokenTopicClient.publish(
+          FGA_CACHE_1,
+          FGA_TOPIC_2,
+          'This weird trick will make you a better developer...'
+        )
+      ).toBeInstanceOf(TopicPublish.Success);
```

Comment:
idk if these 2 publish calls really do anything for the test?

Reply:
They are just creating existing topics on a cache. One of the tests tries to publish to both existing and new topics and ensures both fail when you don't have topic permissions.

Reply:
topics aren't a resource that actually exists or has state on the server though, so I don't think these lines are needed.

Reply:
Removed in https://github.com/momentohq/client-sdk-javascript/pull/598/commits/f2aad35500cbec9f6c42ca858e70abafffacbf58